ndctl/list: display region caps for any of BTT, PFN, DAX

JIRA: https://issues.redhat.com/browse/RHEL-80436

commit def79df43e763dd89973e0732dd49ee5f38416ac
Author: Donet Tom <donettom@linux.vnet.ibm.com>
Date:   Thu Feb 20 00:20:29 2025 -0600

    ndctl/list: display region caps for any of BTT, PFN, DAX
    
    If any one of BTT, PFN, or DAX is not present, but the other two
    are, then the region capabilities are not displayed in the
    ndctl list -R -C command.
    
    This is because util_region_capabilities_to_json() returns NULL
    if any one of BTT, PFN, or DAX is not present.
    
    In this patch, we have changed the logic to display all the region
    capabilities that are present.
    
    Test Results with CONFIG_BTT disabled
    =====================================
    Without this patch
    ------------------
     # ./ndctl list -R -C
     [
      {
        "dev":"region1",
        "size":12884901888,
        "align":16777216,
        "available_size":11257511936,
        "max_available_extent":9630121984,
        "type":"pmem",
        "iset_id":14748366918514061582,
        "persistence_domain":"unknown"
      },
    
    With this patch
    ---------------
     # ./ndctl list -R -C
     [
      {
        "dev":"region1",
        "size":12884901888,
        "align":16777216,
        "available_size":11257511936,
        "max_available_extent":9630121984,
        "type":"pmem",
        "iset_id":14748366918514061582,
        "capabilities":[
          {
            "mode":"fsdax",
            "alignments":[
              65536,
              2097152,
              1073741824
            ]
          },
          {
            "mode":"devdax",
            "alignments":[
              65536,
              2097152,
              1073741824
            ]
          }
        ],
        "persistence_domain":"unknown"
      },
    
    Fixes: 965fa02e372f ("util: Distribute 'filter' and 'json' helpers to per-tool objects")
    Signed-off-by: Donet Tom <donettom@linux.vnet.ibm.com>
    Reviewed-by: Li Zhijian <lizhijian@fujitsu.com>
    Tested-by: Li Zhijian <lizhijian@fujitsu.com>
    Reviewed-by: Jeff Moyer <jmoyer@redhat.com>
    Reviewed-by: Alison Schofield <alison.schofield@intel.com>
    Link: https://lore.kernel.org/r/20250220062029.9789-1-donettom@linux.vnet.ibm.com
    Signed-off-by: Alison Schofield <alison.schofield@intel.com>

Signed-off-by: Jeff Moyer <jmoyer@redhat.com>

diff --git a/ndctl/json.c b/ndctl/json.c
index 23bad7f..7646882 100644
--- a/ndctl/json.c
+++ b/ndctl/json.c
@@ -381,7 +381,7 @@ struct json_object *util_region_capabilities_to_json(struct ndctl_region *region
 	struct ndctl_pfn *pfn = ndctl_region_get_pfn_seed(region);
 	struct ndctl_dax *dax = ndctl_region_get_dax_seed(region);
 
-	if (!btt || !pfn || !dax)
+	if (!btt && !pfn && !dax)
 		return NULL;
 
 	jcaps = json_object_new_array();
